block-producer/src/batch_builder/batch.rs

```diff
@@ -108,12 +110,14 @@ impl TransactionBatch {
             .map(|(account_id, account_states)| (*account_id, account_states.initial_state))
     }
 
-    /// Returns an iterator over (account_id, new_state_hash) tuples for accounts that were
+    /// Returns an iterator over (account_id, details, new_state_hash) tuples for accounts that were
     /// modified in this transaction batch.
-    pub fn updated_accounts(&self) -> impl Iterator<Item = (AccountId, Digest)> + '_ {
-        self.updated_accounts
-            .iter()
-            .map(|(account_id, account_states)| (*account_id, account_states.final_state))
+    pub fn updated_accounts(
+        &self
+    ) -> impl Iterator<Item = (AccountId, Option<AccountDetails>, Digest)> + '_ {
```

Comment:
Can you add a struct type for the return tuple? It should make the code a bit easier to read.